sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClient.java

```diff
@@ -0,0 +1,184 @@
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.amqp.RetryOptions;
+import com.azure.core.http.rest.IterableResponse;
+import com.azure.core.implementation.annotation.ReturnType;
+import com.azure.core.implementation.annotation.ServiceClient;
+import com.azure.core.implementation.annotation.ServiceMethod;
+import com.azure.messaging.eventhubs.implementation.ConnectionOptions;
+import com.azure.messaging.eventhubs.models.EventHubConsumerOptions;
+import com.azure.messaging.eventhubs.models.EventHubProducerOptions;
+import com.azure.messaging.eventhubs.models.EventPosition;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * The main point of interaction with Azure Event Hubs, the client offers a connection to a specific Event Hub within
+ * the Event Hubs namespace and offers operations for sending event data, receiving events, and inspecting the connected
+ * Event Hub.
+ *
+ * <p>
+ * <strong>Creating a synchronous {@link EventHubClient} using an Event Hub instance connection string</strong>
+ * </p>
+ *
+ * {@codesnippet com.azure.messaging.eventhubs.eventhubclient.instantiation}
+ *
+ * @see EventHubClientBuilder
+ * @see EventHubAsyncClient To communicate with Event Hub using an asynchronous client.
+ * @see <a href="https://docs.microsoft.com/Azure/event-hubs/event-hubs-about">About Azure Event Hubs</a>
+ */
+@ServiceClient(builder = EventHubClientBuilder.class)
+public class EventHubClient implements Closeable {
+    private final EventHubAsyncClient client;
+    private final RetryOptions retry;
+    private final EventHubProducerOptions defaultProducerOptions;
+    private final EventHubConsumerOptions defaultConsumerOptions;
+
+    EventHubClient(EventHubAsyncClient client, ConnectionOptions connectionOptions) {
+        Objects.requireNonNull(connectionOptions);
+
+        this.client = Objects.requireNonNull(client);
+        this.retry = connectionOptions.retry();
+        this.defaultProducerOptions = new EventHubProducerOptions()
+            .retry(connectionOptions.retry());
+        this.defaultConsumerOptions = new EventHubConsumerOptions()
+            .retry(connectionOptions.retry())
+            .scheduler(connectionOptions.scheduler());
+    }
+
+    /**
+     * Retrieves information about an Event Hub, including the number of partitions present and their identifiers.
+     *
+     * @return The set of information for the Event Hub that this client is associated with.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public EventHubProperties getProperties() {
+        return client.getProperties().block(retry.tryTimeout());
+    }
+
+    /**
+     * Retrieves the identifiers for all the partitions of an Event Hub.
+     *
+     * @return The identifiers for all partitions of an Event Hub.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public IterableResponse<String> getPartitionIds() {
```

Comment:
Jonathan changed this to `IterableStream` yesterday. You may have to fix this before merging.

Reply:
Ah yes. He hasn't merged it in yet tho. :D Ahaha, going to beat him to the punch!

Reply:
Booooooo!

Reply:
You should have just checked it in with IterableStream. It would be a good example of eventually consistent code.